src/essreflectometry/amor/orso.py

```diff
@@ -1,60 +1,95 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
-import platform
-from datetime import datetime
+"""ORSO utilities for Amor."""
+import numpy as np
+import scipp as sc
+from orsopy.fileio import base as orso_base
+from orsopy.fileio import data_source as orso_data_source
+from orsopy.fileio.orso import Column, Orso, OrsoDataset
 
-from orsopy import fileio
+from ..orso import OrsoDataSource, OrsoInstrument, OrsoIofQDataset, OrsoReduction
+from ..types import NormalizedIofQ1D, QResolution, Run, ThetaData, WavelengthData
 
-from .. import __version__
 
+def build_orso_instrument(
+    events_in_wavelength: WavelengthData[Run], events_in_theta: ThetaData[Run]
+) -> OrsoInstrument[Run]:
+    """Build ORSO instrument metadata from intermediate reduction results for Amor.
 
-def make_orso(
-    owner: fileio.base.Person,
-    sample: fileio.data_source.Sample,
-    creator: fileio.base.Person,
-    reduction_script: str,
-) -> fileio.orso.Orso:
+    This assumes specular reflection and sets the incident angle equal to the computed
+    scattering angle.
     """
-    Generate the base Orso object for the Amor instrument.
-    Populate the Orso object for metadata storage.
-
-    Parameters
-    ----------
-    owner:
-        The owner of the data set, i.e. the main proposer of the measurement.
-    sample:
-        A description of the sample.
-    creator:
-        The creator of the reduced data, the person responsible for the
-        reduction process.
-    reduction_script:
-        The script or notebook used for reduction.
-
-    Returns
-    -------
-    :
-        Orso object with the default parameters for the Amor instrument.
-    """
-    orso = fileio.orso.Orso.empty()
-    orso.data_source.experiment.probe = 'neutrons'
-    orso.data_source.experiment.facility = 'Paul Scherrer Institut'
-    orso.data_source.measurement.scheme = 'angle- and energy-dispersive'
-    orso.reduction.software = fileio.reduction.Software(
-        'scipp-ess', __version__, platform.platform()
+    wavelength = events_in_wavelength.coords['wavelength']
+    incident_angle = events_in_theta.coords['theta']
+    # Explicit conversions to float because orsopy does not like np.float* types.
+    return OrsoInstrument(
+        orso_data_source.InstrumentSettings(
+            wavelength=orso_base.ValueRange(
+                min=float(wavelength.min().value),
+                max=float(wavelength.max().value),
+                unit=_ascii_unit(wavelength.unit),
+            ),
+            incident_angle=orso_base.ValueRange(
+                min=float(incident_angle.min().value),
+                max=float(incident_angle.max().value),
+                unit=_ascii_unit(incident_angle.unit),
+            ),
+            polarization=None,  # TODO how can we determine this from the inputs?
+        )
+    )
+
+
+def build_orso_iofq_dataset(
```

Comment:
Which part here is Amor specific?

Reply:
Knowing which columns to save. I guess that is not so much amor-specific but workflow-specific. Do we have an established pattern for handling this? Should the providers simply not be part of a `providers` tuple?

Reply:
Do we have any reason to believe the workflows at ESS will be different?

Reply:
I don't know how many different types of workflows we will have. Will the only thing we ever save to ORSO be I(Q)?